.github/workflows/code-style.yaml

```diff
@@ -0,0 +1,26 @@
+# Run code style checks on each pull request.
+name: Code style checks
+
+on: [pull_request]
+
+jobs:
+  linting:
+    env:
+      # Configure a constant location for the uv cache
+      UV_CACHE_DIR: /tmp/.uv-cache
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Pixi Environment
+        uses: prefix-dev/setup-pixi@v0.8.1
+        with:
+          pixi-version: v0.25.0
+          cache: true
+          cache-write: ${{ github.event_name == 'push' && github.ref_name == 'main' }}
+
+      - name: Set up Python
+        run: pixi global install python
+
+      - uses: pre-commit/action@v2.0.0
```

Comment:
I don't remember why, but something with this action I found to be really limiting so I prefer to simply call pre-commit in a `run:`, but this is fine for now.

Reply:
I think I remember why. I'm not sure this was it, but it might be this:

I like using `no-commit-to-branch` so that I don't accidentally work on `main`, and make all changes via PRs. But then this test ends up failing in CI. As a workaround, I want to set the environment variable `SKIP=no-commit-to-branch`. I was doing this via a Bash command prefix.

On second thought, I should be able to do this with a step-level `env:`:

```yaml
- uses: pre-commit/action@v2.0.0
  env:
    SKIP: "no-commit-to-branch"
```

Funny that I didn't notice this earlier.

BTW, the current version is `v3.0.1`, not `v2.0.0`. This should definitely be changed!

I'd actually propose sticking with [this job](https://github.com/pre-commit/action?tab=readme-ov-file#using-this-action) for now, since I don't think there's any benefit to uv/pixi here. Indeed, `setup-python` leverages the tool cache on the runner, so in most cases it should be much faster and simpler than using any downloaded Python.